src/ssl/support.h

```diff
@@ -330,7 +330,8 @@ void InRamCertificateDbKey(const Ssl::CertificateProperties &certProperties, SBu
 
 /**
   \ingroup ServerProtocolSSLAPI
-  Generates an OpenSSL BIO for writting to an SBuf object
+  Creates and returns an OpenSSL BIO object for writing to `buf` (or throws).
```

Comment:
If we are going to scope with PR with cleanup (shuffling the method statics around) then the BIO_new_SBuf function itself should be made static internal to the support.cc - it is not used elsewhere and the shuffle would remove one more OpenSSL-specific API definition.

Reply:
`BIO_new()` functions are frequently used outside of ssl/support.cc. I suspect that some of those existing calls and some future calls should be BIO_new_SBuf(). It is possible, of course, that the code surrounding those calls should _itself_ be moved/hidden somewhere, but doing so without GnuTLS/etc. pressure/support would be premature IMO.

There is nothing wrong with OpenSSL-specific code, including APIs, unless there exist a parallel GnuTLS/etc. code/APIs. When both appear, we should keep the high-level code simpler by hiding library-specific details behind common APIs. Until then, we should keep the code simpler by using library-specific details as needed.